src/condor_tools/ping.cpp

```diff
@@ -68,6 +313,106 @@ void process_err_stack(CondorError *errstack) {
 
 }
 
+void print_security_config(bool include_soap, const char * subsys)
+{
+	const char * const knobs[] = {
+		"SEC_%s_AUTHENTICATION_METHODS", "SEC_%s_CRYPTO_METHODS",
+		"SEC_%s_AUTHENTICATION", "SEC_%s_NEGOTIATION",
+		"ALLOW_%s", "DENY_%s",
+	};
+
+	std::set<std::string> reported; // so we don't repeat ourselves
+	std::string param_name;
+	const char * banner = nullptr;
+
+	bool legacy = param_boolean("LEGACY_ALLOW_SEMANTICS", false);
+	if (legacy) { printf("WARNING: LEGACY_ALLOW_SEMANTICS = true\n"); }
+
+	for (auto knob : knobs) {
+		bool is_allow_knob = false;
+		if (starts_with(knob, "ALLOW_")) {
+			banner = "ALLOW";
+			is_allow_knob = true;
+		} else if (starts_with(knob, "DENY_")) {
+			banner = "DENY";
+			is_allow_knob = true;
+		} else {
+			banner = knob + sizeof("SEC_%s");
+		}
+
+		for (DCpermission it = ALLOW_PERM; it < LAST_PERM; it = NEXT_PERM(it)) {
+			// iterate so we do DEFAULT first, and skip ALLOW and SOAP
+			DCpermission perm = it;
+			if (it == DEFAULT_PERM) continue;
+			else if (perm == ALLOW_PERM) perm = DEFAULT_PERM;
+			else if (perm == SOAP_PERM && !include_soap) continue;
+
+			formatstr(param_name, knob, PermString(perm));
+			if (subsys) { param_name += "_"; param_name += subsys; }
+
+			for (auto tag = perm; tag < LAST_PERM; tag = DCpermissionHierarchy::nextConfig(tag, legacy)) {
+				std::string from_param, dotted_param;
+				auto_free_ptr value(SecMan::getSecSetting(knob, tag, &from_param, subsys));
+				if (value) {
+					bool already_reported = (reported.count(from_param) > 0);
+					bool skip_this = (already_reported && !is_allow_knob);
+					if (skip_this) continue;
+					const char* hash = skip_this ? "#" : "";
+
+					if (banner) {
+						printf("\n #--- %s ---#\n", banner);
+						banner = nullptr;
+					}
+
+					printf("%s%s = %s\n", hash, param_name.c_str(), value.ptr());
+					if (param_name != from_param) fprintf(stdout, "%s # from: %s\n", hash, from_param.c_str());
+					reported.insert(from_param);
+
+					const MACRO_META * pmet = nullptr;
+					const char * def_val = nullptr;
+					const char * rawval = param_get_info(from_param.c_str(), subsys, nullptr, dotted_param, &def_val, &pmet);
```

Comment:
Is this used anywhere?

Reply:
the MACRO_META data is used to get file and line number.